static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js

```diff
@@ -49,6 +56,20 @@ define([
         });
     }
 
+    function otherBannersVisible () {
+        var siteMessage = document.getElementsByClassName('js-site-message');
+        if (siteMessage.length && !siteMessage[0].classList.contains('is-hidden')) {
+            // Contribution banner is visible
+            return true;
+        }
+        var breakingNews = document.getElementsByClassName('js-breaking-news-placeholder');
+        if (breakingNews.length && !breakingNews[0].classList.contains('breaking-news--hidden')) {
+            // Breaking news alert is visible
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
Fantastic comment! You've finally pointed out the madness of banners. So the logic to detect them is fine because they're all using `ui/message` but the race condition remains.

I claim that this bit of code is the last to run, because it has to asynchronously download react and make an API request before it can run, but I'll change the code to listen to those events.

Thanks